### PR TITLE
Entity controller update

### PR DESCRIPTION
**Pull request by AMatutat**

### CODE REVIEW

File: core/src/de/fhbielefeld/pmdungeon/vorgaben/game/Controller/EntityController.java [hunk 1]
```diff
@@ -21,11 +22,14 @@ public EntityController() {
      * removes entity if deletable is set true
      */
     public void update() {
+        ArrayList<IEntity> toRemove = new ArrayList();
         for (IEntity obj : dungeonEntitys) {
-            if (obj.deleteable()) removeEntity(obj);
+            if (obj.deleteable()) toRemove.add(obj);
             else obj.update();
         }
+        dungeonEntitys.removeAll(toRemove);
```
Comment (cagix) [inline core/src/de/fhbielefeld/pmdungeon/vorgaben/game/Controller/EntityController.java:30]: @AMatutat  why not simply `dungeonEntitys.removeIf(o -> o.deleteable());`?

Comment (cagix) [reply core/src/de/fhbielefeld/pmdungeon/vorgaben/game/Controller/EntityController.java:30]: btw, you could refactor `dungeonEntitys` to `dungeonEntities` 😎

Comment (AMatutat) [reply core/src/de/fhbielefeld/pmdungeon/vorgaben/game/Controller/EntityController.java:30]: > @AMatutat why not simply `dungeonEntitys.removeIf(o -> o.deleteable());`?

than i would have to iterate twice, once for update and once in removeIf

> btw, you could refactor dungeonEntitys to dungeonEntities 😎

thats true

Comment (AMatutat) [reply core/src/de/fhbielefeld/pmdungeon/vorgaben/game/Controller/EntityController.java:30]:
```    
public void update() {
dungeonEntities.removeIf(obj -> obj.deleteable());
dungeonEntities.forEach(obj -> obj.update());
 }
``` 

feels like haskell now!

Comment (cagix) [reply core/src/de/fhbielefeld/pmdungeon/vorgaben/game/Controller/EntityController.java:30]: @AMatutat little bit more Haskell'ish:

```java
dungeonEntities = dungeonEntities..stream()
        .map(IEntity::update)
        .filter(IEntity::notDeleteable)
        .collect(toList());
```

**Pull request merged**